### ISSUE TITLE
Event issues that effect EventHelper#display_event_status_by_dept issues

### ISSUE THREAD

**Issue by masukomi** [ReadyResponder/ReadyResponder#668]
event.rb: this is misleading

```
  def unavailabilities
    Availability.overlapping(start_time..end_time).active.unavailable
  end
```

and as a result 

`unavailable_people` is  a lie, because they aren't necessarily unavailable. they may be partially available but have a short period during which they're unavailable.

and that leads to a misleading count in the event-status table (  grid of number boxes) when viewing an event they get counted as unavailable when they should be counted as available.

`Event#partially_available_people`
uses `Availability#partially_overlapping` but that's a bug because it doesn't account for people who's availability begins before the start time and ends before the end time. 

this should use `Availability#overlapping` 
Also, I think `Availability#partially_overlapping` should be deleted because i can't see a use case for that particular query.

**Comment by goldenhub77** [comment]
@masukomi @kgf my PR #692 fixes the bugs listed in this issue. Having that said event.rb and availability.rb still need to be refactored in another PR. I plan on reviewing these models to refactor broken/redundant methods in my next PR.

**Comment by kgf** [comment]
Fixed by #692
